zero_bin/common/src/pre_checks.rs

```diff
@@ -0,0 +1,24 @@
+use proof_gen::proof_types::GeneratedBlockProof;
+
+pub fn check_previous_proof_and_checkpoint(
+    checkpoint_block_number: u64,
+    previous_proof: &Option<GeneratedBlockProof>,
+    start: u64,
+) -> anyhow::Result<()> {
+    if let Some(proof) = previous_proof {
+        if proof.b_height + 1 != start {
+            return Err(anyhow::Error::msg(format!(
+                "Previous proof block height {} does not match current starting block height {}",
+                proof.b_height, start,
+            )));
+        }
+    }
+    if checkpoint_block_number >= start {
```

Comment:
If there is no previous proof, we should check that the checkpoint height connects with the current block interval

Reply:
(This was the original comment being this PR, and also something I said twice in previous comments that got mistakenly resolved)

Reply:
If you don't want this to be addressed, then I'd appreciate some justification.

Reply:
> If there is no previous proof, we should check that the checkpoint height connects with the current block interval

@temaniarpit27 @Nashtare I guess we were a bit confusing in our requests. It goes down to:

if previous block proof was not provided,  `checkpoint_block_number` must be `start_block-1`, right? No other way `checkpoint_block_height` could be connected to the interval?

Reply:
in rpc cli, we dont pass previous proof, only `checkpoint_block_number`, so in this case it can be only be `start_block-1`.
in leader binary, we can pass both, previous proof and checkpoint block number
in this case if previous proof if present, checkpoint block number can be less than start block, not necessarily 1 less
if previous proof is not present, then checkpoint block number has to be 1start_block-1`
right?

Reply:
> If you don't want this to be addressed, then I'd appreciate some justification.

https://github.com/0xPolygonZero/zk_evm/pull/517#discussion_r1727470551
i read that only if else needs to be fixed, so i marked it as resolved
apologies for that

Reply:
Source of confusion is probably the rpc cli arguments. Previous block proof is not actually used in the rpc fetch code (we just take `checkpoint_state_trie_root` directly from checkpoint block), but later in the proving code.

Reply:
thanks @atanmarko for clearing the confusion